backend/src/channels/channels.controller.spec.ts

```diff
@@ -19,9 +28,72 @@ describe('ChannelsController', () => {
     }).compile();
 
     controller = module.get<ChannelsController>(ChannelsController);
+    service = module.get<ChannelsService>(ChannelsService);
   });
 
   it('should be defined', () => {
     expect(controller).toBeDefined();
   });
+
+  describe('findAll()', () => {
+    it('should return an array of channels', async () => {
+      const expected = new PageDto(
+        [],
+        new PageMetaDto(new PageOptionsDto(), 10),
+      );
+      jest.spyOn(service, 'findAll').mockImplementation(async () => expected);
+      const result = await controller.findAll();
+      expect(result).toBe(expected);
+    });
+  });
+
+  describe('create()', () => {
+    it('should return a ResponseChannelDto', async () => {
+      const mockChannel = new Channel();
+      mockChannel.name = 'A';
+      mockChannel.type = ChannelType.PUBLIC;
+      const createChannelDto = new CreateChannelDto();
+      createChannelDto.name = 'A';
+      createChannelDto.type = ChannelType.PUBLIC;
+      jest.spyOn(service, 'create').mockImplementation(async () => mockChannel);
+      const result = await controller.create(createChannelDto);
+      expect(result).toEqual(mockChannel);
+    });
+  });
+
+  describe('update()', () => {
+    it('should return a ResponseChannelDto', async () => {
+      const mock = new UpdateResult();
+      const channelDto = new UpdateChannelDto();
+      channelDto.name = 'user';
+      jest.spyOn(service, 'update').mockImplementation(async () => mock);
+      const result = await controller.update('1', channelDto);
+      expect(result).toEqual(mock);
+    });
+  });
+
+  describe('findOne()', () => {
+    it('should return a channel', async () => {
+      const mockOut = new Channel();
+      const expected = new ResponseChannelDto(mockOut[0]);
```

Comment:
```suggestion
      const expected = new ResponseChannelDto();
```
I think that, as ResponseChannelDto extends ChannelDto and this last class does not have a constructor, a default one is constructed by Javascript that ignores extraneous parameters.